### PR TITLE
Course pull request

### PR DESCRIPTION
**Pull request by sandrus88**

### CODE REVIEW

File: src/main/java/org/sg/campus/backing/CourseController.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+public class CourseController {
+
+	@ManagedProperty(value = "#{applicationBean}")
+	private ApplicationBean applicationBean;
+	private TopicController topicController;
+	private List<Course> courseList = new ArrayList<Course>();
+	private Course selectedCourse;
+
+	private List<Topic> allTopics = new ArrayList<Topic>();
+//	private List<Topic> selectedTopics;
```
Comment (ermalaliraj) [inline src/main/java/org/sg/campus/backing/CourseController.java]: remove commented code

File: src/main/java/org/sg/campus/backing/CourseController.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+		List<Topic> checkedTopics = new ArrayList<Topic>();
+		for (int i = 0; i < allTopics.size(); i++) {
+			final Topic topic = allTopics.get(i);
+			if (topic.isChecked()) {
+				checkedTopics.add(topic);
+			}
+		}
+		selectedCourse.setTopicList(checkedTopics);
+		System.out.println(selectedCourse);
+		cleanCheckboxes();
```
Comment (ermalaliraj) [inline src/main/java/org/sg/campus/backing/CourseController.java]: why you should clean the checkboxes after the save? Better to do it only when you abandon the page. Like this you keep looking the selected checkboxes even after the save.

File: src/main/java/org/sg/campus/domain/Topic.java [hunk 3]
```diff
@@ -24,8 +25,15 @@ public void setDescription(String description) {
...
+		return checked;
+	}
+	public void setChecked(boolean checked) {
+		this.checked = checked;
+	}
+	    
 	@Override
 	public String toString() {
-		return "Topic Id: " + id + ", Nome: " + name + ", Description: " + description;
+		return this.getClass().getSimpleName() + " [id: " + id + ", name: " + name + ", description: " + description + ", c: " + checked + "]";
```
Comment (ermalaliraj) [inline src/main/java/org/sg/campus/domain/Topic.java]: instead of "c" put "checked"

File: src/main/webapp/app/course/courseTopicsList.xhtml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+				<h:column>
+					<f:facet name="header">Selection</f:facet>
+					<h:selectBooleanCheckbox id="selectionTopic" value="#{topic.checked}"/>
+		 		</h:column>
+		</h:dataTable>
+		
+		<h:panelGrid id="panelButtonsUpdateTopic" border="0" columns="2">
+			<h:commandButton id="aggiornaTopic" value="Update Topic" action="#{courseController.updateCourseTopics()}" />
+			<h:commandButton id="tornaIndietro" value="Back" action="#{courseController.cleanTopic(selectedCourse)}" />
```
Comment (ermalaliraj) [inline src/main/webapp/app/course/courseTopicsList.xhtml]: what is selectedCourse? In this iteration the n-th topic is "topic"

Comment (ermalaliraj) [inline src/main/webapp/app/course/courseTopicsList.xhtml]: In the cleanTopic you shouldn't pass anything. As the name indicates you should only clean allTopics from the "checked" ones.  In fact better to change the name in: cleanAllTopics()

**Review by ermalaliraj — CHANGES_REQUESTED** [review]

File: src/main/java/org/sg/campus/backing/CourseController.java (hunk 2 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/backing/CourseController.java]: done

File: src/main/java/org/sg/campus/domain/Topic.java (hunk 3 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/domain/Topic.java]: done

File: src/main/webapp/app/course/courseTopicsList.xhtml (hunk 4 above)
Comment (sandrus88) [reply src/main/webapp/app/course/courseTopicsList.xhtml]: done

Comment (sandrus88) [reply src/main/webapp/app/course/courseTopicsList.xhtml]: done

File: src/main/java/org/sg/campus/backing/CourseController.java (hunk 1 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/backing/CourseController.java]: done

File: src/main/java/org/sg/campus/backing/CourseController.java (hunk 2 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/backing/CourseController.java]: done

File: src/main/java/org/sg/campus/domain/Topic.java (hunk 3 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/domain/Topic.java]: done

Comment (sandrus88) [reply src/main/java/org/sg/campus/domain/Topic.java]: done

File: src/main/webapp/app/course/courseTopicsList.xhtml (hunk 4 above)
Comment (sandrus88) [reply src/main/webapp/app/course/courseTopicsList.xhtml]: done

File: src/main/java/org/sg/campus/backing/CourseController.java [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+	}
+	
+	public void cleanCheckboxes() {
```
Comment (ermalaliraj) [inline src/main/java/org/sg/campus/backing/CourseController.java:43]: unused method.  or remove this method,  or call this method from celanTopic() and remove the code there.  In fact cleanTopic() is more like a goBack() where isndie :
1. clean topic
2. return the navigation string

File: src/main/java/org/sg/campus/backing/CourseController.java [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+		}
+	}
+	
+	public String cleanTopic(Course course) {
+		selectedCourse = course;
+		for (int i = 0; i < allTopics.size(); i++) {
+			final Topic topic = allTopics.get(i);
+			topic.setChecked(false);
+		}
+		System.out.println(selectedCourse);
```
Comment (ermalaliraj) [inline src/main/java/org/sg/campus/backing/CourseController.java:55]: I would put a reference in the printed string like: allTopics: (print topics)

File: src/main/java/org/sg/campus/backing/CourseController.java [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+	public void updateCourseTopics() {
+		List<Topic> checkedTopics = new ArrayList<Topic>();
+		for (int i = 0; i < allTopics.size(); i++) {
+			final Topic topic = allTopics.get(i);
+			if (topic.isChecked()) {
+				checkedTopics.add(topic);
+			}
+		}
+		selectedCourse.setTopicList(checkedTopics);
+		System.out.println(selectedCourse);
```
Comment (ermalaliraj) [inline src/main/java/org/sg/campus/backing/CourseController.java:40]: On this method we are changing the topics of the selectedCourse. Better to print only the list of checked and info to chich course belonds. Something like this: :  "slectedTopics for course id:" +  checkedTopics

File: src/main/java/org/sg/campus/backing/CourseController.java (hunk 5 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/backing/CourseController.java:43]: done

File: src/main/java/org/sg/campus/backing/CourseController.java (hunk 6 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/backing/CourseController.java:55]: done

File: src/main/java/org/sg/campus/backing/CourseController.java (hunk 7 above)
Comment (sandrus88) [reply src/main/java/org/sg/campus/backing/CourseController.java:40]: done

**Pull request merged**